src/Laravel/ApiPlatformProvider.php

```diff
@@ -1389,4 +1374,54 @@ public function boot(): void
 
         $this->loadRoutesFrom(__DIR__.'/routes/api.php');
     }
+
+    private function tagServices(): void
+    {
+        $directory = app_path();
+        $classes = ReflectionClassRecursiveIterator::getReflectionClassesFromDirectories([$directory]);
```

Comment:
don't you think we should serialize or cache the tree to avoid possible performance problems?